sycl/plugins/hip/pi_hip.cpp

```diff
@@ -1320,10 +1320,13 @@ pi_result hip_piDeviceGetInfo(pi_device device, pi_device_info param_name,
         hipDeviceGetAttribute(&constant_memory,
                               hipDeviceAttributeTotalConstantMemory,
                               device->get()) == hipSuccess);
-    cl::sycl::detail::pi::assertion(constant_memory >= 0);
 
+    // hipDeviceGetAttribute takes a int*, however the size of the constant
+    // memory on AMD GPU may be larger than what can fit in the positive part
+    // of a signed integer, so we need to cast it to unsigned before returning
+    // the value.
     return getInfo(param_value_size, param_value, param_value_size_ret,
-                   pi_uint64(constant_memory));
+                   pi_uint64(static_cast<unsigned int>(constant_memory)));
```

Comment:
can't you just make the `constant_memory` be `unsigned int`?

Reply:
The problem is `hipDeviceGetAttribute` takes a `int *`, so we do need a cast, either here, or I can make `constant_memory` a `unsigned int` and cast the pointer when calling `hipDeviceGetAttribute`.

Reply:
>  make constant_memory a unsigned int and cast the pointer when calling hipDeviceGetAttribute

Let's do so. Not sure the cast would be really needed.

Reply:
Updated as suggested, however a cast is still needed, without it complains with the following:
```
llvm/sycl/plugins/hip/pi_hip.cpp:1320:31: error: invalid conversion from ‘unsigned int*’ to ‘int*’ [-fpermissive]
         hipDeviceGetAttribute(&constant_memory,
                               ^~~~~~~~~~~~~~~~
```